projects/plugins/jetpack/modules/woocommerce-analytics/classes/class-jetpack-woocommerce-analytics-trait.php

```diff
@@ -285,6 +341,72 @@ public function get_user_id() {
 		return 'null';
 	}
 
+	/**
+	 * Gets the IDs of additional blocks on the Cart/Checkout pages or templates.
+	 *
+	 * @param string $cart_or_checkout Whether to get blocks on the cart or checkout page.
+	 * @return array All inner blocks on the page.
+	 */
+	public function get_additional_blocks_on_page( $cart_or_checkout = 'cart' ) {
+
+		$additional_blocks_on_page_transient_name = 'jetpack_woocommerce_analytics_additional_blocks_on_' . $cart_or_checkout . '_page';
+		$additional_blocks_on_page                = get_transient( $additional_blocks_on_page_transient_name );
+
+		if ( false !== $additional_blocks_on_page ) {
+			return $additional_blocks_on_page;
+		}
+
+		$content = $this->cart_content_source;
+
+		if ( 'checkout' === $cart_or_checkout ) {
+			$content = $this->checkout_content_source;
+		}
+
+		$parsed_blocks = parse_blocks( $content );
+		$other_blocks  = array_filter(
+			$parsed_blocks,
+			function ( $block ) use ( $cart_or_checkout ) {
+				if ( ! isset( $block['blockName'] ) ) {
+					return false;
+				}
+				if ( 'woocommerce/classic-shortcode' === $block['blockName'] ) {
+					return false;
+				}
+				if ( 'core/shortcode' === $block['blockName'] ) {
+					return false;
+				}
+				if ( 'checkout' === $cart_or_checkout && 'woocommerce/checkout' !== $block['blockName'] ) {
+					return true;
+				}
+				if ( 'cart' === $cart_or_checkout && 'woocommerce/cart' !== $block['blockName'] ) {
+					return true;
+				}
+				return false;
```

Comment:
Nitpic: I believe adding a few line breaks here will increase the readability.

```suggestion
				if ( ! isset( $block['blockName'] ) ) {
					return false;
				}

				if ( 'woocommerce/classic-shortcode' === $block['blockName'] ) {
					return false;
				}
				
				if ( 'core/shortcode' === $block['blockName'] ) {
					return false;
				}
				
				if ( 'checkout' === $cart_or_checkout && 'woocommerce/checkout' !== $block['blockName'] ) {
					return true;
				}
				
				if ( 'cart' === $cart_or_checkout && 'woocommerce/cart' !== $block['blockName'] ) {
					return true;
				}
				
				return false;
```